src/classes/Utility/AdminShippingTabDataProvider.php

```diff
@@ -0,0 +1,229 @@
+            'name' => $shippingMethod ? $shippingMethod->getTitle() : '',
+            'reference' => $shipmentDetails->getReference(),
+            'deleted' => $shipmentDetails->isDeleted(),
+            'icon' => $shippingMethod ? $shippingMethod->getLogoUrl() : '',
+            'status' => $shipmentDetails->getShippingStatus() ?: '',
+            'time' => $shipmentDetails->getLastStatusUpdateTime() !== null
+                ? $timeProvider->serializeDate($shipmentDetails->getLastStatusUpdateTime(), 'd.m.Y H:i:s')
+                : '',
+            'carrier_tracking_numbers' => $shipmentDetails->getCarrierTrackingNumbers(),
+            'carrier_tracking_url' => $shipmentDetails->getCarrierTrackingUrl() ?: '',
+            'packlink_shipping_price' => $shipmentDetails->getShippingCost() !== null
+                ? $shipmentDetails->getShippingCost() . ' €' : '',
+            'link' => self::getOrderDraftUrl($shipmentDetails->getReference()),
+        );
+    }
+
+    /**
+     * Prepares a template for displaying and printing labels.
+     *
+     * @param OrderShipmentDetails $shipmentDetails Shipment details for an order.
+     */
+    private static function prepareLabelsTemplate(OrderShipmentDetails $shipmentDetails)
+    {
+        self::$context->controller->addJS(
+            self::$module->getPathUri() . 'views/js/PrestaPrintShipmentLabels.js?v=' . self::$module->version,
+            false
+        );
+
+        /** @var OrderService $orderService */
+        $orderService = ServiceRegister::getService(OrderService::CLASS_NAME);
+
+        $labels = $shipmentDetails->getShipmentLabels();
+        $status = $shipmentDetails->getStatus() ?: ShipmentStatus::STATUS_PENDING;
+
+        $isLabelPrinted = !empty($labels) && $labels[0]->isPrinted();
+
+        self::$context->smarty->assign(
+            array(
+                'orderId' => $shipmentDetails->getOrderId(),
+                'isLabelPrinted' => $isLabelPrinted,
+                'date' => !empty($labels) ? $labels[0]->getDateCreated()->format('d/m/Y') : '',
+                'status' => TranslationUtility::__(
+                    $isLabelPrinted ? 'Printed' : 'Ready'
+                ),
+                'isLabelAvailable' => !empty($labels) || $orderService->isReadyToFetchShipmentLabels($status),
+                'number' => '#PLSL1',
+            )
+        );
+    }
+
+    /**
+     * Returns link to order draft on Packlink for the provided shipment reference.
+     *
+     * @param string $reference Shipment reference.
+     *
+     * @return string Link to order draft on Packlink.
+     */
+    private static function getOrderDraftUrl($reference)
+    {
+        /** @var Configuration $configService */
+        $configService = ServiceRegister::getService(Configuration::CLASS_NAME);
+        $userInfo = $configService->getUserInfo();
+
+        $userDomain = 'com';
+        if ($userInfo !== null && in_array($userInfo->country, array('ES', 'DE', 'IT', 'FR'))) {
```

Comment:
Why don't you use constants in 'ES','DE','IT','FR'?

Reply:
I don't see the need for it. This is the only location where this is needed in this class.

Reply:
to improve the code if you use constants you could use the same constants in several files , in this case you are using this Strings in two diferrents files

Reply:
refactored and moved to the Core.